### PR TITLE
Feature/92 nav scenario

### PR DESCRIPTION
**Pull request by thibaudteil**
* **Tickets addressed:** bsk-092
* **Review:** By commit
* **Merge strategy:** Merge (no squash)  

## Description
Modifications stemming from a navigation scenario linking modules on develop. 
Contains validity check modifications, regularization of point cloud representation, improvements to read a set of images from file, and additional checks of message content validity. 

## Verification
Unit test suite still passes and was adjusted to the changes

## Documentation
Documentation still holds

## Future work
None

### CODE REVIEW

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp:85]: :large_orange_diamond: Should we update the member variable's comment in the header to say e.g. "Number of times called with valid input data..."

File: src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h]: :large_orange_diamond: We lost the default initializer syntax with this edit '{}'. Was that desired?

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h
Comment (patkenneally) [inline src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h]: :large_orange_diamond: Same here on the default initializer.

**Review by patkenneally — APPROVED** [review]
Legend:
:small_red_triangle_down: *Issues* to address before merge
:large_orange_diamond: *Requests* that should not block merge, but should at least be discussed
:large_blue_circle: *Recommendations* that can be ignored if desired

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.cpp:85]: Good idea!

File: src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/cameraTriangulation/cameraTriangulation.h]: I fixed a bunch of these

File: src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h
Comment (thibaudteil) [reply src/fswAlgorithms/pointCloudProcessing/pointCloudTriangulation/pointCloudTriangulation.h]: Same as above

**Pull request merged**
